docs/source/spec-workload-executor.rst

```diff
@@ -43,29 +43,98 @@ After accepting the inputs, the workload executor:
-   * MUST run operations sequentially and in the order in which they appear in the ``operations`` array.
-   * MUST repeat the entire set of specified operations indefinitely, until the **termination signal** from
-     ``astrolabe`` is received.
-   * MUST keep count of the number of operations failures (``numFailures``) that are encountered while running
-     operations. An operation failure is when the actual return value of an operation does not match its
-     expected return value (as defined in the ``result`` field of the ``driverWorkload``).
-   * MUST keep count of the number of operation errors (``numErrors``) that are encountered while running
-     operations. An operation error is when running an operation unexpectedly raises an error. Workload executors
-     implementations should try to be as resilient as possible to these kinds of operation errors.
-   * MUST keep count of the number of operations that are run successfully (``numSuccesses``).
+     
+#. MUST use the driver's unified test runner to execute the tests in the
+   scenario, and the operations in each test, sequentially and in the order
+   in which they appear in the ``tests`` and ``operations`` arrays,
+   with the following deviations from the unified test runner specification:
+   
+   * The workload executor MUST repeat execution of the entire set of
+     specified tests and operations indefinitely, until the
+     **termination signal** from ``astrolabe`` is received.
+   
+   * The workload executor MUST keep count of the number of failures
+     (``numFailures``) that are encountered. A failure is when
+     the actual return value of an operation or a published event does not match
+     the respective expected return value or event (as these expectations
+     are defined in the unified test format).
+   
+   * The workload executor MUST record all errors encountered while running the scenario.
+     An operation error is any error that is propagated out of the unified test runner.
+     Workload executor implementations should try to be as resilient
+     as possible to these kinds of operation errors.
+   
+   * The workload executor MUST keep count of the number of operation errors (``numErrors``) that
+     are encountered while running the scenario.
+   
+   * The workload executor MUST keep count of the number of invocations of the scenario that
+     did not result in an error (``numSuccesses``).
 
 #. MUST set a signal handler for handling the termination signal that is sent by ``astrolabe``. The termination signal
    is used by ``astrolabe`` to communicate to the workload executor that it should stop running operations. Upon
    receiving the termination signal, the workload executor:
 
    * MUST stop running driver operations and exit soon.
-   * MUST dump collected workload statistics as a JSON file named ``results.json`` in the current working directory
+   * MUST write the collected events and errors into a JSON file named
```

Comment:
Can we clarify that exact signals that must be caught for POSIX and Windows? That would codify things in this document and avoid having folks consult the `utils.py` source.

Reply:
I think that's a reasonable request but I do believe it is outside the scope of this PR, especially given that I haven't used astrolabe on windows at all and am not too familiar with the details. Could you make a new issue at https://github.com/mongodb-labs/drivers-atlas-testing/issues/new to cover this please?

Reply:
Resolving per @prashantmital's comment that this is already documented.